### ISSUE TITLE
Required attachments

### ISSUE THREAD

**Issue by yckart** [nicbell/attach.js#2]
It would nice to have the possibility to add *dependent* attachments.

Sure, we could do this simply like so:
```js
Attach.add('pluginName1', function(el) {
  if (~el.dataset.attach.indexOf('pluginName2')) {
    // stuff
  }
})
```

...however, wouldn't it nice to have something like this and let it managed internally?
```js
Attach.add('pluginName1', ['pluginName2'], function(el) {
  // stuff...
})
```

**Comment by mrmartineau** [comment]
@yckart I agree. I have been using Attach on a project recently and have often found issues with loading and instantiation where one attachment depends on another, or if one manipulates the DOM onLoad/DOMContentReady and another is skipped because it sits within it.. (don't think that makes sense, butI think this sort of interface might help.

**Comment by nicbell** [comment]
@mrmartineau in situations where you have several modules that depend on each other I would write another module that governs their creation and any other inter-dependent logic. I would attach this module to the parent container instead of attaching all the child modules. 

For example if you had several types of filters on a page that triggered each other. You could keep those filters single purpose and create a `FilterManager` module to take care of the depended business logic between the filters.

**Comment by yckart** [comment]
Let's assume that we want to add an editor to some sort of input field and in some cases also a toolbar. So, this toolbar requires that the editor is attached, as the editor give's us access to the current selection or what ever.

This might be not the best example, but would be a case where it could helpful to have this feature available.
